static/app/views/eventsV2/table/columnEditCollection.tsx

```diff
@@ -136,22 +142,71 @@ class ColumnEditCollection extends React.Component<Props, State> {
     this.props.onChange([...this.props.columns, newColumn]);
   };
 
-  handleUpdateColumn = (index: number, column: Column) => {
+  handleUpdateColumn = (index: number, updatedColumn: Column) => {
     const newColumns = [...this.props.columns];
-    if (column.kind === 'equation') {
+    if (updatedColumn.kind === 'equation') {
       this.setState(prevState => {
         const error = new Map(prevState.error);
-        error.set(index, parseArithmetic(column.field).error);
+        error.set(index, parseArithmetic(updatedColumn.field).error);
         return {
           ...prevState,
           error,
         };
       });
+    } else {
+      // Update any equations that contain the existing column
+      this.updateEquationFields(newColumns, index, updatedColumn);
     }
-    newColumns.splice(index, 1, column);
+    newColumns.splice(index, 1, updatedColumn);
     this.props.onChange(newColumns);
   };
 
+  updateEquationFields = (newColumns: Column[], index: number, updatedColumn: Column) => {
```

Comment:
Shouldn't this be `oldColumns`?
```suggestion
  updateEquationFields = (oldColumns: Column[], index: number, updatedColumn: Column) => {
```

Reply:
I wanted to match the naming in `handleUpdateColumn` which calls it `newColumns` as well

Reply:
It's called `newColumns` in `handleUpdateColumn` because it's a copy of the old columns that are going to be mutated. But since you passed the value prior to it being mutated, this still contains the old columns right? Not a big deal but just thought it was a little confusing.

Reply:
Aye, but these newColumns are being mutated here too right? that's why I figured matching the names would be more correct. I'm willing to update this though, but going to merge this for now.